### PR TITLE
Properly log websocket errors

### PR DESCRIPTION
**Pull request by vladsud**

### CODE REVIEW

File: packages/drivers/driver-base/src/documentDeltaConnection.ts [hunk 1]
```diff
@@ -511,11 +511,16 @@ export class DocumentDeltaConnection
...
         // Note: we suspect the incoming error object is either:
         // - a string: log it in the message (if not a string, it may contain PII but will print as [object Object])
-        // - a socketError: add it to the OdspError object for driver to be able to parse it and reason
-        //   over it.
+        // - an Error object thrown by socket.io engine. Be careful with not recording PII!
         let message = `socket.io: ${handler}`;
-        if (typeof error === "string") {
+        if (typeof error !== "object") {
             message = `${message}: ${error}`;
+        } else if (error.type === "TransportError") {
```
Comment (vladsud) [inline packages/drivers/driver-base/src/documentDeltaConnection.ts:518]: See here: https://github.com/socketio/engine.io-client/blob/master/lib/transport.js#L30

**Comment by vladsud** [comment]
ping

File: packages/drivers/driver-base/src/documentDeltaConnection.ts [hunk 2]
```diff
@@ -511,11 +511,17 @@ export class DocumentDeltaConnection
...
-        let message = `socket.io: ${handler}`;
-        if (typeof error === "string") {
+        // - an Error object thrown by socket.io engine. Be careful with not recording PII!
+        let message = `socket.io (${handler})`;
+        if (typeof error !== "object") {
             message = `${message}: ${error}`;
+        } else if (error?.type === "TransportError") {
+            // Websocket errors reported by engine.io-client.
+            // They are Error objects with description containing WS error and description = "TransportError"
+            // Please see https://github.com/socketio/engine.io-client/blob/master/lib/transport.js#L30,
```
Comment (DLehenbauer) [inline packages/drivers/driver-base/src/documentDeltaConnection.ts:521]: Link was 404 for me.  Perhaps:
```
https://github.com/socketio/engine.io-client/blob/master/lib/transport.ts#L44
```

**Review by DLehenbauer — APPROVED** [review]

File: packages/drivers/driver-base/src/documentDeltaConnection.ts (hunk 2 above)
Comment (wes-carlson) [reply packages/drivers/driver-base/src/documentDeltaConnection.ts:521]: Maybe one of these would be more stable as opposed to linking to master:
https://github.com/socketio/engine.io-client/blob/7245b803e0c8d57cfc1f1cd8b8c8d598e8397967/lib/transport.ts#L44
https://github.com/socketio/engine.io-client/blob/3.5.0/lib/transport.js#L69

**Pull request merged**